### ISSUE TITLE
♦️ Gemas permitidas ♦️

### ISSUE THREAD

**Issue by jtvaldivia** [IIC2143/Syllabus#116]
Hola! 
Aca les dejo una lista con gemas permitidas para el desarrollo de su proyecto 😁, si quieren usar una que no esté, hagan un comentario indicando su nombre y la razón de por qué quieren usarla. Actualizare a lista a medida que pregunten🧐.

## Gemas permitidas
* `devise`
* `cancancan`
* `rolify`
* `will_paginate`
* `faker`
* `rails-i18n`
* `devise-i18n`
* `rspec-rails`
* `rails-controller-testing`
* `faker`
* `factory_bot_rails`
* `annotate`
* `awesome_print`
* `simplecov`
* `omniauth-google-oauth2`
* `jquery-ui-rails`
* `geocoder`
* `cloudinary`
* `bulmajs-rails`
* `commontator`
* `redis`
* `excon`
* `chartkick`
* `simple_form`
* `kaminari`
* `figaro`
* `fog-aws`
* `aws-sdk-s3`
* `active_storage`
* `active_storage_validations`
* `image-processing`
* `rails_admin`
* `pundit`
* `bootstrap`
* `dotenv-rails`
* `carrierwave`
* `date_validator`
* `aasm`

## Gemas prohibidas
* `jumpstart`
* `suspender`
* `cometchat-pro`
* `ransack`
* `gravatar`
* `solidus`
* `activeadmin`

**Comment by k0rval4n** [comment]
Se puede utilizar TailwindCSS?

**Comment by jtvaldivia** [comment]
@k0rval4n, si pueden ocupar TailwindCSS, tambien bootstrap, bulma o css.

**Comment by MathiasPrett** [comment]
Hola. Podemos ocupar la gema [Ricecream](https://github.com/nodai2hITC/ricecream) ?? 🙏

**Comment by jtvaldivia** [comment]
@MathiasPrett si pueden

**Comment by esperanzaacevedo** [comment]
Hola!, se puede ocupar Shrine?

**Comment by jtvaldivia** [comment]
@esperanzaacevedo  si pueden, igual hay otras gemas que hacen cosas parecidas por si no les funciona.

**Comment by michebordoni** [comment]
hola, se puede usar Followability para las solicitudes de amistad/seguidores?

**Comment by jtvaldivia** [comment]
@michebordoni si pueden

**Comment by jtvaldivia** [comment]
@michebordoni depende, si las solicitudes son para el modelo, no. Debido a las implicancia que podria tener. En caso que sea como para otro tipo de solicitud si pueden

**Comment by michebordoni** [comment]
> @michebordoni depende, si las solicitudes son para el modelo, no. Debido a las implicancia que podria tener. En caso que sea como para otro tipo de solicitud si pueden

con modelo a que te refieres? la queremos usar para que los users se puedan seguir/ser amigos entre si

**Comment by jtvaldivia** [comment]
@michebordoni  a entonces si pueden ocuparla, me referia como a los modelos minimos que pide el proyecto por enunciado.

**Comment by DaniSZ02** [comment]
hola, se puede usar la gema "whenever" ?

**Comment by jtvaldivia** [comment]
@DaniSZ02  si pueden.

**Comment by SantiagoDLCB** [comment]
Hola! La gema rails_admin que actualmente esta permitida instala automaticamenta las gemas 
gem "sassc-rails"
gem 'paper_trail'

Asumo que no es problema cierto?

**Comment by FraMqueli** [comment]
Hola, se podrá usar la gema Turbo Streams? es para eso de la actualización automática de los mensajes, para no tener que estar actualizando la pagina cada vez que llega un msj nuevo.

**Comment by endlessloop2** [comment]
Hola, se puede usar `rack_cors`? Esto para permitir ciertas requests necesarias para hacer funcionar nuestro chat.

En todo caso, asumiría que si ya que esa gema viene incluida (pero comentada) en el Gemfile de los proyectos creados con --api.

**Comment by dgolzm** [comment]
Hola! Se puede usar la gema `bourbon`?

**Comment by jtvaldivia** [comment]
Hola,
@FraMqueli si pueden

**Comment by jtvaldivia** [comment]
@endlessloop2 Si pueden mientras tengan cuidado como la ocupen, deben tener ojo con als request y donde se llamen

**Comment by jtvaldivia** [comment]
@dgolzm, si pueden pero les recomendaria mas ocupar bulma en ese caso.

**Comment by endlessloop2** [comment]
Hola, se puede activar la gema de `redis`? Es necesaria para ActionCable en producción para el chat.

**Comment by FelipeMunozMoraga** [comment]
@endlessloop2 Si se puede

**Comment by 1982739** [comment]
Se puede ocupar la gema twilio-ruby para el chat?

**Comment by jtvaldivia** [comment]
@1982739 Si pueden

**Comment by k0rval4n** [comment]
Se puede utilizar libvips? Es requisito de active_storage.

**Comment by jtvaldivia** [comment]
@k0rval4n si pueden

**Comment by mmunozv** [comment]
Hola se puede usar `mocha` para setear current_user en los tests?

**Comment by jtvaldivia** [comment]
@mmunozv  pueden mientras no ocupen Rspec en el proceso.

**Comment by Alenachoxd** [comment]
se puede usar la gema minitest-reporters?

**Comment by esperanzaacevedo** [comment]
hola la gema dotenv-rails, esta permitida?

**Comment by jtvaldivia** [comment]
@Alenachoxd para que quieren ocuparla?

**Comment by jtvaldivia** [comment]
@esperanzaacevedo si pueden, esta en la lista de permitida.

**Comment by Alenachoxd** [comment]
> @Alenachoxd para que quieren ocuparla?

Para la visualizacion del ouput de la consola al hacer rails test

**Comment by jtvaldivia** [comment]
@Alenachoxd  si pueden si es para eso.

**Comment by SantiagoDLCB** [comment]
Hola! Se puede usar minimagick para transformar imágenes, antes de subirlas usando carrierwave y cloudinary?

**Comment by su-hoxton** [comment]
Hola, podemos ocupar la gema [rainbow](https://github.com/ku1ik/rainbow) para modificar los puts de la consola del servidor?
Y la gema [email_validator](https://github.com/K-and-R/email_validator) para manejar la regex de validaciones?

**Comment by maneschwarz** [comment]
Hola, se puede ocupar noticed?

**Comment by laurapalacios01** [comment]
Hola! se puede ocupar sidekiq?

**Comment by sbarriag** [comment]
Hola se puede usar [Turbo](https://github.com/Deanout/turbochat) para el chat?

**Comment by jtvaldivia** [comment]
@su-hoxton  si pueden ocupar email_validator si es para ver validaciones.
Rainbow con que objetivo lo quieren ocupar?

**Comment by jtvaldivia** [comment]
@maneschwarz quieren ocuparlo para el chat?

**Comment by jtvaldivia** [comment]
@laurapalacios01  con que objetivo la quieren ocupar?

**Comment by jtvaldivia** [comment]
@sbarriag no, pueden ocupar otras gemas pero tienen que preguntar para saber cual esta permitida. Hay varias para chat que no dejamos ocupar

**Comment by su-hoxton** [comment]
@jtvaldivia Rainbow lo queremos ocupar para loggear checkpoints en la consola del servidor de forma que sea más fácil distinguirlos de los llamados a la base de datos y los endpoints que la consola muestra automáticamente

**Comment by jtvaldivia** [comment]
@su-hoxton ya si ocupenlo no mas

**Comment by maneschwarz** [comment]
> @maneschwarz quieren ocuparlo para el chat?

Queremos ocupar notice para hacer notificaciones.

**Comment by icortesc** [comment]
> @mmunozv pueden mientras no ocupen Rspec en el proceso.

Pero esta permitido ocupar RSpec o no? Pregunto porque en la lista del inicio dice que se puede.

**Comment by k0rval4n** [comment]
> @sbarriag no, pueden ocupar otras gemas pero tienen que preguntar para saber cual esta permitida. Hay varias para chat que no dejamos ocupar

Hola! No se puede usar Turbo con Action Cable? Es para que los mensajes se actualicen al recibirlos.

**Comment by k0rval4n** [comment]
Aprovecho de preguntar si se puede usar importmap y stimulus para el manejo de javascripts.

**Comment by jtvaldivia** [comment]
@icortesc  sale en el enunciado que deben ocupar minitest.

**Comment by jtvaldivia** [comment]
@k0rval4n importmap no pueden ocuparlo.Stimulus si pueden. Turbo pueden mientras sea soloo para actualizar

**Comment by Segismundios** [comment]
Holaa, se puede usar webpacker??

**Comment by jtvaldivia** [comment]
@Segismundios  si pueden ocuparlo, pero ojo que es una gema que generalmente se ocupa para rails 6 o mas bajo

**Comment by k0rval4n** [comment]
> @Segismundios si pueden ocuparlo, pero ojo que es una gema que generalmente se ocupa para rails 6 o mas bajo

Cuál sería la forma correcta de manejar JS entonces? Como dice, en Rails 7 se reemplazó webpack por importmap.

**Comment by jtvaldivia** [comment]
@k0rval4n  pueden ocupar webpacker,  esbuild y sprocket. Si no,  importmap lo pueden ocupar mientras no lo ocupen para aplicar otros frameworks como react.

**Comment by Allende30** [comment]
hola, se puede usar letter opener? para poder ver los mails que se mandan

**Comment by jtvaldivia** [comment]
@Allende30  si pueden. Pero ojo que para la presentacion queda mejor mostrarlo desde un correo
